File: duplicate-zeros/duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        int[] result = new int[arr.length];
        int resultCounter=0;
        for(int i = 0; i<arr.length && resultCounter<=arr.length-1; i++){
            if(arr[i]!=0){
                result[resultCounter]=arr[i];
                resultCounter++;
            }else{
                if(resultCounter<arr.length){
                    result[resultCounter]=0;
                }
                if(resultCounter+1<arr.length){
                    result[resultCounter+1]=0;
                }
                resultCounter +=2;
            }
        }
        for(int i=0; i<arr.length; i++){
            arr[i]=result[i];
        }
    }
}

File: 1168-duplicate-zeros/duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        j = 0
        ans = []
        while(j < len(arr)):
            if arr[j] == 0:
                ans.append(0)
                if len(ans) == len(arr):
                    j+=1
                    break
                ans.append(0)
                j+=1
            else:
                ans.append(arr[j])
                j+=1
            
        for index in range(len(arr)):
            arr[index] = ans[index]